engines/net.c

```diff
@@ -1110,12 +1167,49 @@ static int fio_netio_setup_connect_unix(struct thread_data *td,
 	return 0;
 }
 
+#ifdef CONFIG_VSOCK
+static int fio_netio_setup_connect_vsock(struct thread_data *td,
+					const char *host, unsigned short port)
+{
+	struct netio_data *nd = td->io_ops_data;
+	struct sockaddr_vm *addr = &nd->addr_vm;
+	int cid;
+
+	if (!host) {
+		log_err("fio: connect with no host to connect to.\n");
+		if (td_read(td))
+			log_err("fio: did you forget to set 'listen'?\n");
+
+		td_verror(td, EINVAL, "no hostname= set");
+		return 1;
+	}
+
+	addr->svm_family = AF_VSOCK;
+	addr->svm_port = port;
+
+	if (host) {
+		cid = atoi(host);
+		if (cid < VMADDR_CID_LOCAL || cid > UINT32_MAX) {
```

Comment:
fuzzy CI is failing here. I guess because we introduced `VMADDR_CID_LOCAL` in Linux v5.6 and in some way that job as an old header. BTW I think we can just check if the CID is positive, since also 0 is allowed (but not implemented for now at least in virtio-vsock).

```suggestion
		if (cid < 0 || cid > UINT32_MAX) {
```